pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -269,6 +270,7 @@ public void addSegment(String segmentName, TableConfig tableConfig, IndexLoading
       }
       _logger.info("Initialize RealtimeSegmentDataManager - " + segmentName);
       _segmentDataManagerMap.put(segmentName, manager);
+      _serverMetrics.addValueToTableGauge(_tableNameWithType, ServerGauge.SEGMENT_COUNT, 1L);
```

Comment:
This one should be added inside the else block in line 237, or we will double counting because this gauge will also be updated in `addSegment(ImmutableSegmentLoader.load(indexDir, indexLoadingConfig, schema));`

Reply:
Yes， this is inside this else block, just the last line :)

Reply:
Oh, yes you are right haha